src/aligned_memory.rs

```diff
@@ -0,0 +1,75 @@
+//! Aligned memory
+
+/// Provides u8 slices at a specified alignment
+#[derive(Clone, Debug, PartialEq)]
+pub struct AlignedMemory {
+    len: usize,
+    align_offset: usize,
+    write_index: usize,
+    mem: Vec<u8>,
+}
+impl AlignedMemory {
+    /// Return a new AlignedMem type
+    pub fn new(len: usize, align: usize) -> Self {
```

Comment:
Maybe we can avoid using `align_offset` by dividing the allocation in chunks like this instead:
`let mem = vec![AlignedChunk::default(); (len + std::mem::size_of::<AlignedChunk>() - 1) / std::mem::size_of::<AlignedChunk>()];`

```
#[derive(Default)]
#[repr(C, align(16))]
struct AlignedChunk([u8; 16]); // Maybe even [u8; 1], not sure how the padding behaves here
```

Of course that would require a type reinterpretation or iterator flattening at the outer interfaces.

Reply:
Yeah, not sure the advantages outweigh the complexities